File: Chapter10-Pachyderm-Language-Clients/create-pipeline.go
package main

 import (
     "log"
     "fmt"

     "github.com/pachyderm/pachyderm/v2/src/client"
     "github.com/pachyderm/pachyderm/v2/src/pps"
)

func main() {

     c, err := client.NewOnUserMachine("user")
     if err != nil {
         log.Fatal(err)
     }

     if err := c.CreatePipeline(
         "contour",
         "svekars/contour-histogram:1.0 ",
         []string{"python3", "/contour.py"},
         []string{},
         &pps.ParallelismSpec{
             Constant: 1,
         },
         client.NewPFSInput("photos", "/"),
         "",
         false,
     ); err != nil {
         panic(err)
     }

     if err := c.CreatePipeline(
         "histogram",
         "svekars/contour-histogram:1.0",
         []string{"python3", "/histogram.py"},
         []string{},
         &pps.ParallelismSpec{
             Constant: 1,
         },
         client.NewPFSInput("contour", "/"),
         "",
         false,
     ); err != nil {
         panic(err)
     }

     pipelines, err := c.ListPipeline(true)
     if err != nil {
         panic(err)
     }
     fmt.Println(pipelines)
 }

File: Chapter10-Pachyderm-Language-Clients/create-pipeline.py
import python_pachyderm
from python_pachyderm.service import pps_proto

client = python_pachyderm.Client()
client.create_pipeline(
    pipeline_name="contour",
    transform=pps_proto.Transform(
        cmd=["python3", "contour.py"],
        image="svekars/contour-histogram:1.0",
    ),
    input=pps_proto.Input(
        pfs=pps_proto.PFSInput(glob="/", repo="photos")
    ),
)
client.create_pipeline(
    pipeline_name="histogram",
    transform=pps_proto.Transform(
        cmd=["python3", "histogram.py"],
        image="svekars/contour-histogram:1.0",
    ),
    input=pps_proto.Input(
        pfs=pps_proto.PFSInput(glob="/", repo="contour")
    ),
)
print(list(client.list_pipeline()))
